### PR TITLE
CASMINST-6949: create k8s auditing volumes properly

### PR DESCRIPTION
**Pull request by bo-quan**
# Description
When k8s api server auditing is turned on, two volume mounts need to be added to the config. The original script didn't handle existing volume mounts which caused the k8s upgrade to fail. This PR provides the fix for the issue.

# Checklist

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

### CODE REVIEW

**Review by kimjensen-hpe — APPROVED** [review]

**Review by leliasen-hpe — APPROVED** [review]

File: upgrade/scripts/k8s/upgrade_control_plane.sh [hunk 1]
```diff
@@ -45,7 +45,17 @@ cm_auditing_enabled=$?
...
-  sed -i '/    apiServer:/a\      extraVolumes:\n      - hostPath: /var/log/audit/kl8s/apiserver\n        mountPath: /var/log/audit/kl8s/apiserver\n        name: k8s-audit-log\n        pathType: DirectoryOrCreate\n        readOnly: false\n      - hostPath: /etc/kubernetes/audit\n        mountPath: /etc/kubernetes/audit\n        name: k8s-audit\n        pathType: DirectoryOrCreate\n        readOnly: true' /tmp/kubeadm-config.yaml
+  # a temporary replacement hack for the subsequent yq4 command to work
+  sed -i 's/ClusterConfiguration: |/ClusterConfiguration:/' /tmp/kubeadm-config.yaml
+  if [ "$(yq4 eval '.data.ClusterConfiguration.apiServer.extraVolumes' /tmp/kubeadm-config.yaml)" == null ]; then
+    # No existing volumes
+    sed -i '/    apiServer:/a\      extraVolumes:\n      - hostPath: /var/log/audit/kl8s/apiserver\n        mountPath: /var/log/audit/kl8s/apiserver\n        name: k8s-audit-log\n        pathType: DirectoryOrCreate\n        readOnly: false\n      - hostPath: /etc/kubernetes/audit\n        mountPath: /etc/kubernetes/audit\n        name: k8s-audit\n        pathType: DirectoryOrCreate\n        readOnly: true' /tmp/kubeadm-config.yaml
```
Comment (rustydb) [inline upgrade/scripts/k8s/upgrade_control_plane.sh:52]: Replace fragile `sed` command with `yq4`.
If you need help let me know.

**Review by rustydb — CHANGES_REQUESTED** [review]
There are several fragile `sed` commands in this file, we should try and replace them as we maintain these workflows.

File: upgrade/scripts/k8s/upgrade_control_plane.sh (hunk 1 above)
Comment (bo-quan) [reply upgrade/scripts/k8s/upgrade_control_plane.sh:52]: @rustydb I understand that it's nice to have, but I ran into some issue trying to do the replacement. Could we merge it as it is?

Comment (rustydb) [reply upgrade/scripts/k8s/upgrade_control_plane.sh:52]: I'll submit changes in a follow-up PR.

**Review by rustydb — APPROVED** [review]

**Pull request merged**